### DISCUSSION TITLE
Blob

### DISCUSSION THREAD

**Discussion by punkcanyang** [IntensiveCoLearning/Web3-URL#214] [Q&A] [answered]
我看了文件说0.8.24之后才支援blob存储
这样Ethstorage也支援blob吗？

另外一个合约中哪些是存到链下的Storage，哪些是存在链上呢？

**Comment by qzhodl** [comment]
ethfs-cli 中 -t 这个参数，1 代表使用calldata上传+以太坊原生存储， 2代表使用BLOB上传+ethstorage永久存储

合约里面 这个[函数](https://github.com/ethstorage/evm-large-storage/blob/ebeb30bba804d3199001d52e93eee2d18c940fcf/contracts/ERC5018.sol#L78)对应calldata, 这个[函数](https://github.com/ethstorage/evm-large-storage/blob/ebeb30bba804d3199001d52e93eee2d18c940fcf/contracts/ERC5018.sol#L91)对应blob

现在 sepolia 和 quarkchain l2 testnet 都是支持 BLOB上传+ethstorage永久存储的

**Reply by punkcanyang** [reply]
如果是自己写一个manual合约呢？

另外如果要支援Chunk，是否要引入ERC5018？

**Reply by qzhodl** [reply]
如果是自己实现一个manual mode的合约，接口都可以自由定义的

**Reply by qzhodl** [reply]
如果希望使用 BLOB上传 + ethstorage存储，只需要在自己的合约里面调用 es 的合约即可。[这里](https://github.com/ethstorage/evm-large-storage/blob/ebeb30bba804d3199001d52e93eee2d18c940fcf/contracts/BlobStorageManager.sol#L11) 是 es 的部分接口

**Reply by punkcanyang** [reply]
是调用已经布署的合约吗？还是把BlobStorageManager引入使用？

如果我是要布署在Sepolia或是Holesky上也可以用ethstorage吗？

**Reply by qzhodl** [reply, marked as answer]
Sepolia 上面已经部署了 EthStorage，所以可以直接调用或者引入BlobStorageManager。Holesky 上面还没有部署 EthStorage

**Reply by punkcanyang** [reply]
好的

**Reply by punkcanyang** [reply]
目前我看到StorageManager，LargeStorageManager，BlobStorageManager
假设我不采用Ethstorage，但是要做到可以存入大档案，是不是只有StorageManager可以用？

**Reply by punkcanyang** [reply]
https://holesky.blobscan.com/
Holesky 如果还没有部署 EthStorag，这些Blob的数据是其他的储存?

开始要深入应用之后才发现水很深.....

**Reply by qzhodl** [reply]
这应该是blobscan的服务器自己存储的
